### PR TITLE
Ug and dg

### LINKED ISSUE #58: Code Review for @danielpappa
`src/main/java/cashleh/budget/BudgetHandler.java `
* Line 50: Try to set the values of 0.25 and 1 to a variable, as currently there is little context as to the purpose of these values, so as to reduce usage of magic values

`src/main/java/cashleh/budget/Progress.java`
* Line 48: Since this variable is used as a limit, consider setting this to a constant variable

### PR DESCRIPTION
**Pull request by danielpappa**
Update DG and solve code review issue
closes #58

### CODE REVIEW

**Review by mcmc101001 — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -30,9 +30,42 @@ The **User** sends command which is processed by Input
...
+**Step 1**\
+The printBudget method of BudgetHandler is called.
+
+**Step 2**\
+The BudgetHandler self invokes its own method setBudgetPercentage() to retrieve the data from the
+financial statement.
+
+**Step 3**\
+If the budget is not active (has not been initialised or deleted by the user), a CashLehBudgetException is thrown.
+If the budge is active (has been initialised or updated by the user), the BudgetHandler self-invokes two methods:
```
Comment (WooKaiNing) [inline docs/DeveloperGuide.md]: trivial error: typo in word "budget"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -30,9 +30,42 @@ The **User** sends command which is processed by Input
...
+* printBasicWarning(): checks if the user's budget still provides a good cushion and prints a warning if not
```
Comment (WooKaiNing) [inline docs/DeveloperGuide.md]: may want to give slightly more details on what "good cushion" is ie. percentage of budget progress before warning is triggered?

**Review by WooKaiNing — APPROVED** [review]
diagrams are detailed, code is improved with removal of magic numbers

**Pull request merged**